extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java

```diff
@@ -74,12 +79,28 @@
             }
         }
 
+        // hard-coded support for exposed
+        handleExpose(config, unPrefixed);
+
         result.putAll(unPrefixed);
         result.putAll(quarkusPrefixed);
         result.putAll(toS2iProperties(quarkusPrefixed));
         return result;
     }
 
+    private static void handleExpose(Config config, Map<String, Object> unPrefixed) {
+        for (String generator : EXPOSABLE_GENERATORS) {
+            Optional<Boolean> unprefixedExpose = config.getOptionalValue(generator + "." + EXPOSE_PROPERTY_NAME,
+                    Boolean.class);
+            Optional<Boolean> prefixedExpose = config
+                    .getOptionalValue(QUARKUS_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class);
+            if ((unprefixedExpose.isPresent() && unprefixedExpose.get())
+                    || (prefixedExpose.isPresent() && prefixedExpose.get())) {
+                unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
+            }
+        }
+    }
```

Comment:
```suggestion
    private static void handleExpose(Config config, Map<String, Object> unPrefixed) {
        for (String generator : EXPOSABLE_GENERATORS) {
            boolean unprefixedExpose = config.getOptionalValue(generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class)
                    .orElse(false);
            boolean prefixedExpose = config.getOptionalValue(QUARKUS_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, Boolean.class)
                    .orElse(false);
            if (unprefixedExpose || prefixedExpose) {
                unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
            }
        }
    }
```

To simply the if condition a bit. WDYT? @geoand

Reply:
Updated, thanks